beacon_node/execution_layer/src/engines.rs

```diff
@@ -18,14 +19,70 @@ use types::{Address, ExecutionBlockHash, Hash256};
 const PAYLOAD_ID_LRU_CACHE_SIZE: usize = 512;
 
 /// Stores the remembered state of a engine.
-#[derive(Copy, Clone, PartialEq, Debug)]
+#[derive(Copy, Clone, PartialEq, Debug, Eq, Default)]
 enum EngineState {
     Synced,
+    #[default]
     Offline,
     Syncing,
     AuthFailed,
 }
 
+impl EngineState {
+    /// Returns true if the engine is responsive, regardless of its syncing state; returns false
+    /// otherwise.
+    pub fn is_online(&self) -> bool {
+        match self {
+            EngineState::Synced | EngineState::Syncing => true,
+            EngineState::Offline | EngineState::AuthFailed => false,
+        }
+    }
+}
+
+/// Wrapper structure that ensures changes to the engine state are correctly reported to watchers.
+struct State {
+    /// The actual engine state.
+    state: EngineState,
+    /// Notifier to watch whether the engine is online or not.
+    notifier: watch::Sender<bool>,
+}
+
+impl std::ops::Deref for State {
+    type Target = EngineState;
+
+    fn deref(&self) -> &Self::Target {
+        &self.state
+    }
+}
+
+impl Default for State {
+    fn default() -> Self {
+        let state = EngineState::default();
+        let (notifier, _receiver) = watch::channel(state.is_online());
```

Comment:
What you said is right, however this is what I expect. Since the default state in sync is online, we start normally. If there isn't an execution engine, that state will never be changed. If there is, then the right behaviour is to pause at the beginning. It seems not intuitive but I think it's right